src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const list = [];
+  let listItem = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        listItem = { ...listItem, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        for (const item of action.keysToRemove) {
+          delete listItem[item];
+        }
+        break;
+
+      case 'clear':
+        listItem = {};
+        break;
```

Comment:
The 'clear' action sets `listItem` to an empty object. This is correct, but ensure that this behavior aligns with the task requirements. If the task specifies that 'clear' should reset `listItem` to the initial state instead of an empty object, you should modify this logic accordingly.